File: js/exec01.js
/****** 买卖股票的最佳时机 ******/ 
const maxProfit = function(prices) {
  if(prices.length<2) return 0
  let m = 0;
  for (let i=1;i<prices.length;i++) {
    m += Math.max(0,prices[i] - prices[i-1])
  }
  return m
};
console.log(maxProfit([7,2,3,1,5,6])); // ==> 6

File: go/exec/exec01.go
package exec

import "fmt"

/* 买卖股票的最佳时机 */
var (
	ans int
)

func MaxProfit(prices []int) {
	for i := 1; i < len(prices); i++ {
		ans += clacMax(0, prices[i]-prices[i-1])
	}
	fmt.Println(ans)
}

func clacMax(a, b int) int {
	if b > a {
		return b
	}
	return a
}
